### PR TITLE
Traktor S3: Fix timedelta calculation bugs

### PR DESCRIPTION
**Pull request by ywwg**
jslint fixes (first commit)
Fix incorrect timecode wraparound constant value.
Make time delta clamping better

Fixes https://bugs.launchpad.net/mixxx/+bug/1958925

### CODE REVIEW

File: res/controllers/Traktor-Kontrol-S3-hid-scripts.js [hunk 1]
```diff
@@ -721,16 +721,12 @@ TraktorS3.Deck.prototype.wheelDeltas = function(value) {
 
     if (prevTime > timeval) {
         // We looped around.  Adjust current time so that subtraction works.
-        timeval += 0x100000;
+        timeval += 0x1000000;
```
Comment (ywwg) [inline res/controllers/Traktor-Kontrol-S3-hid-scripts.js:724]: here is the timecode wraparound fix value.  Note that this doesn't protect against the user not touching the jog wheel for a large amount of time (enough for the internal counter to wrap around more than once)

File: res/controllers/Traktor-Kontrol-S3-hid-scripts.js [hunk 2]
```diff
@@ -721,16 +721,12 @@ TraktorS3.Deck.prototype.wheelDeltas = function(value) {
...
-    if (timeDelta === 0) {
-        // Spinning too fast to detect speed!  By not dividing we are guessing it took 1ms.
-        // (This is almost certainly not going to happen on this controller.)
-        timeDelta = 1;
-    }
-
-    var tickDelta = 0;
+    // Clamp Suspiciously low numbers.  Even flicking the wheel as fast as I can, I can't get it
+    // lower than this value.  Lower values (esp below zero) cause huge jumps / errors in playback.
+    const timeDelta = Math.max(timeval - prevTime, 500);
```
Comment (ywwg) [inline res/controllers/Traktor-Kontrol-S3-hid-scripts.js:728]: here's the other change

File: res/controllers/Traktor-Kontrol-S3-hid-scripts.js (hunk 1 above)
Comment (ywwg) [reply res/controllers/Traktor-Kontrol-S3-hid-scripts.js:731]: ok I added wallclock protection

**Comment by JoergAtGithub** [comment]
Why Main and not 2.3?

**Comment by ywwg** [comment]
good call

File: res/controllers/Traktor-Kontrol-S3-hid-scripts.js [hunk 3]
```diff
@@ -235,7 +236,7 @@ TraktorS3.Deck.prototype.defineScaler = function(msg, name, deckOffset, deckBitm
 };
 
 TraktorS3.Deck.prototype.registerInputs = function(messageShort, messageLong) {
-    var deckFn = TraktorS3.Deck.prototype;
+    const deckFn = TraktorS3.Deck.prototype;
```
Comment (uklotzde) [inline res/controllers/Traktor-Kontrol-S3-hid-scripts.js:239]: The `const` keyword is not supported yet in 2.3. Please revert those jslint changes from 2.4.

**Review by uklotzde — CHANGES_REQUESTED** [review]

File: res/controllers/Traktor-Kontrol-S3-hid-scripts.js (hunk 3 above)
Comment (ywwg) [reply res/controllers/Traktor-Kontrol-S3-hid-scripts.js:239]: ah yes ok

**Review by uklotzde — APPROVED** [review]
Ready?

**Review by Swiftb0y — APPROVED** [review]
Changes seem sensible. LGTM. feel free to merge

File: res/controllers/Traktor-Kontrol-S3-hid-scripts.js [hunk 4]
```diff
@@ -1906,8 +1909,8 @@ TraktorS3.Controller.prototype.registerOutputPackets = function() {
     engine.connectControl("[Master]", "maximize_library", TraktorS3.bind(TraktorS3.Controller.prototype.maximizeLibraryOutput, this));
 
     // Master VuMeters
-    this.masterVuMeter["VuMeterL"].connection = engine.makeConnection("[Master]", "VuMeterL", TraktorS3.bind(TraktorS3.Controller.prototype.masterVuMeterHandler, this));
-    this.masterVuMeter["VuMeterR"].connection = engine.makeConnection("[Master]", "VuMeterR", TraktorS3.bind(TraktorS3.Controller.prototype.masterVuMeterHandler, this));
+    this.masterVuMeter.VuMeterL.connection = engine.makeUnbufferedConnection("[Master]", "VuMeterL", TraktorS3.bind(TraktorS3.Controller.prototype.masterVuMeterHandler, this));
```
Comment (uklotzde) [inline res/controllers/Traktor-Kontrol-S3-hid-scripts.js:1912]: The mapping is broken in version 2.3 because this unrelated change slipped through.

https://bugs.launchpad.net/mixxx/+bug/1960680

Comment (Swiftb0y) [reply res/controllers/Traktor-Kontrol-S3-hid-scripts.js:1912]: Sorry. I didnt see that this was a 2.3 PR
fix in #4671
